### PR TITLE
ci: PLT-499: upgrade to python 3.12 and switch to debian base image 

### PR DESCRIPTION
**Pull request by farioas**
Switch to debian base image, leverage more docker layers and finally upgrade to python 3.12 with possibility to adopt 3.13

### CODE REVIEW

**Comment by farioas** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/11827208787)
> Docker image was pushed with the tag `20241113.225528-plt-499-96eccbe99`

**Review by jombooth — APPROVED** [review]

**Comment by farioas** [comment]
/docker build


> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/11843298530)
> Docker image was pushed with the tag `20241114.183146-plt-499-233c10ee5`

File: Dockerfile [hunk 1]
```diff
@@ -37,79 +48,102 @@ RUN --mount=type=cache,target=${YARN_CACHE_FOLDER},sharing=locked \
...
+COPY pyproject.toml poetry.lock README.md ./
+
+# Install dependencies without dev packages
+RUN --mount=type=cache,target=$POETRY_CACHE_DIR,sharing=locked \
+    poetry check --lock && poetry install --no-root --without test --extras "uwsgi"
+
+# Install LS
+COPY label_studio label_studio
+RUN --mount=type=cache,target=$POETRY_CACHE_DIR,sharing=locked \
+    poetry install --only-root --extras "uwsgi" && \
```
Comment (jombooth) [inline Dockerfile]: do we need `--extras uwsgi` in both places?

File: pyproject.toml [hunk 2]
```diff
@@ -225,7 +226,7 @@ requests-mock = "1.12.1"
 twine = ">=4.0.2"
 
 [tool.poetry.extras]
-mysql = ["mysqlclient"]
```
Comment (jombooth) [inline pyproject.toml]: :pouring_liquid:

File: Dockerfile (hunk 1 above)
Comment (farioas) [reply Dockerfile]: yes, because --only-root uninstalls extras https://github.com/python-poetry/poetry/issues/7302

**Review by jombooth — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (jombooth) [reply Dockerfile]: wild, please comment that! and while in there I think you can remove the double quotes

**Pull request closed without merging**